[SPIRV] Restrict OpName generation to major values

s-perron opened the pull request:
Refines OpName emission to only target Global Variables, Functions,
Function Parameters, Local Variables (allocas/phis), and Basic Blocks.
This reduces binary size and clutter by avoiding OpName for every
intermediate instruction (arithmetic, casts, etc.), while preserving
readability for interfaces and program structure.

Also updates the test suite to align with this change:
- Removes OpName checks for intermediate instructions.
- Adds side-effects (e.g., volatile stores) to tests where instructions
  were previously kept alive solely by their OpName usage.
- Updates checks to use generic ID matching where specific names are no
  longer available.
- Adds debug-info/opname-filtering.ll to verify the new policy.

s-perron committed:
[SPIRV] Restrict OpName generation to major values

Refines OpName emission to only target Global Variables, Functions,
Function Parameters, Local Variables (allocas/phis), and Basic Blocks.
This reduces binary size and clutter by avoiding OpName for every
intermediate instruction (arithmetic, casts, etc.), while preserving
readability for interfaces and program structure.

Also updates the test suite to align with this change:
- Removes OpName checks for intermediate instructions.
- Adds side-effects (e.g., volatile stores) to tests where instructions
  were previously kept alive solely by their OpName usage.
- Updates checks to use generic ID matching where specific names are no
  longer available.
- Adds debug-info/opname-filtering.ll to verify the new policy.

s-perron commented:
@michalpaszkowski As discussed in the metting yesterday. Here is a PR. Please add people who had opinions as reviewers. I did not get everyone's name during the call.

Keenuts reviewed (approved):
OK with this change. Wondering if we should have some kind of `-fcgl` flag to disable some optimizations when we know we don't want a necessarily valid module (to avoid the volatile hack)

s-perron commented on llvm/test/CodeGen/SPIRV/extensions/SPV_KHR_float_controls2/decoration.ll:
```diff
@@ -139,101 +93,115 @@ declare dso_local spir_func noundef nofpclass(nan inf) <2 x float> @_Z23__spirv_
 entry:
   %addRes = fadd float %1,  %2
-  store float %addRes, float* @G_addRes
+  store volatile float %addRes, float* @G_addRes
```
They are volatile so that the stores are not removed. since we sometimes write to the same location multiple times, I don't want it removed as a redundant store. We could avoid the volatile if we had a different variable for each store, but I felt that cluttered the test.

s-perron committed:
Merge branch 'main' into opname

MrSidims reviewed (approved):
Is there a way to put old behaviour under a flag? I feel like it's useful for debugging to have LLVM IR to SPIR-V OpName match for intermediate SSA values. But definitely the change makes sense to be default for production.

s-perron committed:
[SPIRV] Add command line option to emit OpName for all values

Added --spirv-emit-op-names option to allow users to force emission of OpName instructions for all values, which is useful for debugging. By default, the conservative behavior of only emitting names for globals, functions, and allocas is preserved to minimize binary size and avoid interfering with optimizations.
